src/mount.ts

```diff
@@ -29,8 +34,8 @@ import { stubComponents } from './stubs'
 
 type Slot = VNode | string | { render: Function }
 
-interface MountingOptions<Props> {
-  data?: () => Record<string, unknown>
+interface MountingOptions<Props, Data = {}> {
+  data?: () => unknown extends Data ? never : Partial<Data>
```

Comment:
How can `unknown` extend `Data` 🤔 ? What is happening here? Doesn't this means we know a bit more about data (eg, it's an `object` for example?)

Reply:
extending object is better 😄   Thank you 👍